process/block/preprocess/transactionsV2.go

```diff
@@ -0,0 +1,416 @@
+			}
+		}
+	}
+
+	if miniBlock, ok := mapMiniBlocks[core.MetachainShardId]; ok {
+		if len(miniBlock.TxHashes) > 0 {
+			miniBlocks = append(miniBlocks, splitMiniBlockIfNeeded(miniBlock, mapSCTxs)...)
+		}
+	}
+
+	return miniBlocks
+}
+
+func splitMiniBlockIfNeeded(miniBlock *block.MiniBlock, mapSCTxs map[string]struct{}) block.MiniBlockSlice {
+	splitMiniBlocks := make(block.MiniBlockSlice, 0)
+	notScTxHashes := make([][]byte, 0)
+	scTxHashes := make([][]byte, 0)
+
+	for _, txHash := range miniBlock.TxHashes {
+		_, isSCTx := mapSCTxs[string(txHash)]
+		if !isSCTx {
+			notScTxHashes = append(notScTxHashes, txHash)
+			continue
+		}
+
+		scTxHashes = append(scTxHashes, txHash)
+	}
+
+	if len(notScTxHashes) > 0 {
+		notScMiniBlock := &block.MiniBlock{
+			TxHashes:        notScTxHashes,
+			SenderShardID:   miniBlock.SenderShardID,
+			ReceiverShardID: miniBlock.ReceiverShardID,
+			Type:            miniBlock.Type,
+			Reserved:        miniBlock.Reserved,
+		}
+
+		splitMiniBlocks = append(splitMiniBlocks, notScMiniBlock)
+	}
+
+	if len(scTxHashes) > 0 {
+		scMiniBlock := &block.MiniBlock{
+			TxHashes:        scTxHashes,
+			SenderShardID:   miniBlock.SenderShardID,
+			ReceiverShardID: miniBlock.ReceiverShardID,
+			Type:            miniBlock.Type,
+			Reserved:        miniBlock.Reserved,
+		}
+
+		splitMiniBlocks = append(splitMiniBlocks, scMiniBlock)
+	}
+
+	return splitMiniBlocks
+}
+
+//TODO: Finish implementation of this method which should create all the scheduled mini blocks if needed
+func (txs *transactions) createScheduledMiniBlocks(
+	haveTime func() bool,
+	isShardStuck func(uint32) bool,
+	isMaxBlockSizeReached func(int, int) bool,
+	sortedTxs []*txcache.WrappedTransaction,
+	mapSCTxs map[string]struct{},
+) block.MiniBlockSlice {
+	log.Debug("createScheduledMiniBlocks has been started")
+
+	mapMiniBlocks := make(map[uint32]*block.MiniBlock)
+
+	for shardID := uint32(0); shardID < txs.shardCoordinator.NumberOfShards(); shardID++ {
+		mapMiniBlocks[shardID] = txs.createEmptyMiniBlock(txs.shardCoordinator.SelfId(), shardID, block.ScheduledBlock)
+	}
+
+	mapMiniBlocks[core.MetachainShardId] = txs.createEmptyMiniBlock(txs.shardCoordinator.SelfId(), core.MetachainShardId, block.ScheduledBlock)
+
+	for index := range sortedTxs {
+		if !haveTime() {
+			log.Debug("time is out in createScheduledMiniBlocks")
+			break
+		}
+
+		txHash := sortedTxs[index].TxHash
+		senderShardID := sortedTxs[index].SenderShardID
+		receiverShardID := sortedTxs[index].ReceiverShardID
+
+		if isShardStuck != nil && isShardStuck(receiverShardID) {
+			log.Trace("shard is stuck", "shard", receiverShardID)
+			continue
+		}
+
+		tx, ok := sortedTxs[index].Tx.(*transaction.Transaction)
+		if !ok {
+			log.Debug("wrong type assertion",
+				"hash", txHash,
+				"sender shard", senderShardID,
+				"receiver shard", receiverShardID)
+			continue
+		}
+
+		if !core.IsSmartContractAddress(tx.RcvAddr) {
```

Comment:
should not use IsSmartContractAddress but destinationTxType.

Reply:
Done